numba/mviewbuf.c

```diff
@@ -10,23 +27,39 @@ static void free_buffer(Py_buffer * buf)
     PyBuffer_Release(buf);
 }
 
+/**
+ * Return a pointer to the data of a writable buffer from obj. If only a
+ * read-only buffer is available and force is True, a read-write buffer based on
+ * the read-only buffer is obtained. Note that this may have some surprising
+ * effects on buffers which expect the data from their read-only buffer not to
+ * be modified.
+ */
 static PyObject*
 memoryview_get_buffer(PyObject *self, PyObject *args){
     PyObject *obj = NULL;
+    PyObject *force = NULL;
     PyObject *ret = NULL;
     void * ptr = NULL;
+    const void* roptr = NULL;
     Py_ssize_t buflen;
     Py_buffer buf;
 
-    if (!PyArg_ParseTuple(args, "O", &obj))
+    if (!PyArg_ParseTuple(args, "O|O", &obj, &force))
         return NULL;
 
-    if (!get_buffer(obj, &buf)) { /* new buffer api */
+    if (!get_buffer(obj, &buf, (force == Py_True))) { /* new buffer api */
```

Comment:
`force == Py_True` isn't great, what if the user passes 1 instead? I think you could simply take an integer argument (which works also with Python bools).